api/devicerepository.proto

```diff
@@ -0,0 +1,388 @@
+  // Application identifiers.
+  ApplicationIdentifiers application_ids = 1 [(gogoproto.customname) = "ApplicationIDs"];
+
+  // Brand identifier, as defined in the Device Repository.
+  string brand_id = 2 [(gogoproto.customname) = "BrandID"];
+  // Model identifier, as defined in the Device Repository.
+  string model_id = 3 [(gogoproto.customname) = "ModelID"];
+
+  // Field mask paths.
+  google.protobuf.FieldMask field_mask = 4 [(gogoproto.nullable) = false];
+}
+
+message ListEndDeviceModelsRequest {
+  // Application identifiers.
+  ApplicationIdentifiers application_ids = 1 [(gogoproto.customname) = "ApplicationIDs"];
+
+  // List end devices from a specific brand.
+  string brand_id = 2 [(gogoproto.customname) = "BrandID"];
+
+  // Limit the number of results per page.
+  uint32 limit = 3 [(validate.rules).uint32.lte = 1000];
+  // Page number for pagination. 0 is interpreted as 1.
+  uint32 page = 4;
+  // Order end devices
+  string order_by = 5 [(validate.rules).string = {
+    in: ["", "brand_id", "-brand_id", "model_id", "-model_id", "name", "-name"]
+  }];
+
+  // List end devices matching a query string.
+  string search = 6;
+
+  // Field mask paths.
+  google.protobuf.FieldMask field_mask = 7 [(gogoproto.nullable) = false];
+}
+
+message GetTemplateRequest {
+  // Application identifiers.
+  ApplicationIdentifiers application_ids = 1 [(gogoproto.customname) = "ApplicationIDs"];
+
+  // End device version information.
+  EndDeviceVersionIdentifiers version_ids = 2 [(gogoproto.customname) = "VersionIDs"];
+}
+
+message GetPayloadFormatterRequest {
+  // Application identifiers.
+  ApplicationIdentifiers application_ids = 1 [(gogoproto.customname) = "ApplicationIDs"];
+
+  // End device version information.
+  EndDeviceVersionIdentifiers version_ids = 2 [(gogoproto.customname) = "VersionIDs"];
+}
+
+message ListEndDeviceBrandsResponse {
+  repeated EndDeviceBrand brands = 1;
+}
+
+message ListEndDeviceModelsResponse {
+  repeated EndDeviceModel models = 1;
+}
+
+message MessagePayloadFormatter {
+  // Payload formatter type.
+  PayloadFormatter formatter = 1 [(validate.rules).enum.defined_only = true];
+  // Parameter for the formatter, must be set together.
+  string formatter_parameter = 2;
+}
+
+service DeviceRepository {
+  rpc ListBrands(ListEndDeviceBrandsRequest) returns (ListEndDeviceBrandsResponse) {
+    option (google.api.http) = {
+      get: "/dr/brands"
+
+      additional_bindings {
+        get: "/dr/applications/{application_ids.application_id}/brands"
+      }
+    };
+  };
+
+  rpc GetBrand(GetEndDeviceBrandRequest) returns (EndDeviceBrand) {
+    option (google.api.http) = {
+      get: "/dr/brands/{brand_id}"
+
+      additional_bindings {
+        get: "/dr/applications/{application_ids.application_id}/brands/{brand_id}"
+      }
+    };
+  }
+
+  rpc ListModels(ListEndDeviceModelsRequest) returns (ListEndDeviceModelsResponse) {
+    option (google.api.http) = {
+      get: "/dr/brands/{brand_id}/models"
```

Comment:
How are we getting `search` in?

Reply:
With a querystring parameter, e.g. `?search=uno`